### PR TITLE
chore: Optimize and clean up Gradle build script

### PR DESCRIPTION
**Pull request by valfirst**
## Change list

- Remove unused declaration
- Optimize Gradle tasks configuration
 
## Types of changes

What types of changes are you proposing/introducing to Java client?

- [x] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -67,12 +67,6 @@ dependencies {
     testRuntimeOnly "org.slf4j:slf4j-simple:${slf4jVersion}"
 }
 
-ext {
-    Sources = fileTree("$buildDir/src/main/java").include('**/*.java')
-    Tests = fileTree("$buildDir/src/test/java").include('**/*.java')
-    Docs = file("$buildDir/doc")
-}
```
Comment (valfirst) [inline build.gradle:74]: unused

File: build.gradle [hunk 2]
```diff
@@ -81,7 +75,7 @@ jacoco {
     toolVersion = '0.8.8'
 }
 
-tasks.withType(JacocoReport) {
+tasks.withType(JacocoReport).configureEach {
```
Comment (valfirst) [inline build.gradle:78]: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_migration_guidelines

File: build.gradle [hunk 3]
```diff
@@ -209,7 +203,7 @@ test {
     finalizedBy jacocoTestReport
 }
 
-task xcuiTest( type: Test ) {
+tasks.register('xcuiTest', Test) {
```
Comment (valfirst) [inline build.gradle:206]: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_migration_guidelines

File: build.gradle [hunk 4]
```diff
@@ -224,7 +218,7 @@ task xcuiTest( type: Test ) {
     }
 }
 
-task uiAutomationTest( type: Test ) {
+tasks.register('uiAutomationTest', Test) {
```
Comment (valfirst) [inline build.gradle:221]: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:task_configuration_avoidance_migration_guidelines

**Review by mykola-mokhnach — APPROVED** [review]
LGTM

**Pull request merged**